lib/types/filesystem.nix

```diff
@@ -24,7 +24,19 @@
     mountpoint = lib.mkOption {
       type = lib.types.nullOr diskoLib.optionTypes.absolute-pathname;
       default = null;
-      description = "Path to mount the filesystem to";
+      description = ''
+        Path to mount the filesystem to during normal operation, or null to
+        disable. The NixOS configuration module will add an entry under
+        `fileSystems` to mount the filesystem at this path.
+      '';
+    };
+    installMountpoint = lib.mkOption {
```

Comment:
Would be nice to also have an impermanence example using this. It's also often what people want to see anyway.